### PR TITLE
[DOC] updated reference for Residual Normalised Score

### LINKED ISSUE #366: Reference for Residual Normalized Score
Hello, I think the reference for the Residual Normalized score is not the good one it should be 

Lei et al., “Distribution-Free Predictive Inference For Regression.”

Instead of the gentle Introduction

### PR DESCRIPTION
**Pull request by mehdi-elion**
# Description

Changed the reference for the residual normalized score from
```
[1] Angelopoulos, A. N., & Bates, S. (2021). A gentle introduction to conformal
prediction and distribution-free uncertainty quantification. arXiv preprint arXiv:2107.07511.
```
to
```
[1] Lei, J., G'Sell, M., Rinaldo, A., Tibshirani, R. J., & Wasserman, L. (2017). Distribution-Free
Predictive Inference For Regression. arXiv:1604.04173.
```

Fixes #366 

## Type of change

This is a simple documentation update.

# How Has This Been Tested?

This has been tested according to the items of the checklist below.

# Checklist

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: doc/theoretical_description_conformity_scores.rst
Comment (vincentblot28) [inline doc/theoretical_description_conformity_scores.rst:112]: It is best to refer to the published version instead of the pre-print : https://www.tandfonline.com/doi/full/10.1080/01621459.2017.1307116

Comment (mehdi-elion) [reply doc/theoretical_description_conformity_scores.rst]: Sure, I'll replace it with a reference to the version you shared in the next commit👍

**Review by vincentblot28 — APPROVED** [review]

**Review by vincentblot28 — COMMENTED** [review]
Thank you very much for this PR ! Could you just update the file history.rst so that we can follow to changes for the next release. Once it's done you'll be able to merge your PR

**Comment by mehdi-elion** [comment]
Sure :) 
I added the following bullet point in `HISTORY.rst`:
```markdown
* Fix reference for residual normalised score in documentation
```

**Review by vincentblot28 — APPROVED** [review]

**Comment by mehdi-elion** [comment]
@vincentblot28 thanks a lot for approving the changes !

I'm sorry for asking but, since I still am quite unfamiliar with MAPIE's merging policy, I'm unsure about whether PRs should be merged by contributors or owners.

Let me know if you need any addition action from me :)

**Comment by vincentblot28** [comment]
@mehdi-elion, your PR is ready, you can merge it ! Thank you for taking in charge this issue

**Comment by mehdi-elion** [comment]
Awesome! Unfortunately I fear that don't have write access / merging rights (yet ?) to MAPIE's repository. 
Are you observing that on your side as well @vincentblot28 ? If so, would it be possible to grant me with sufficient rights to merge the PR :) ?
(If not, I apologize for the misunderstanding, this is my 1st contribution to MAPIE)

**Comment by thibaultcordier** [comment]
> Awesome! Unfortunately I fear that don't have write access / merging rights (yet ?) to MAPIE's repository. Are you observing that on your side as well @vincentblot28 ? If so, would it be possible to grant me with sufficient rights to merge the PR :) ? (If not, I apologize for the misunderstanding, this is my 1st contribution to MAPIE)

Don't worry :) do you see the green "Merge pull request" button below this comment? If so, you have the right to finish the pull request. If not, no problem, I'll merge the pull request for you straight away.

**Comment by thibaultcordier** [comment]
> @thibaultcordier I actually can't see the green "Merge pull request" button at the end of this thread ^^' Here's a screenshot of what I see (just in case). It looks like I'll have to rely on you to have it merged :)

No problem, I'll merge your pull request. Sorry for the wait and thanks again :)

**Pull request merged**
